### DISCUSSION TITLE
Testing payload passed within the request - testing

### DISCUSSION THREAD

**Discussion by mdczaplicki** [aio-libs/aiohttp#8060] [Q&A]
I'm creating `TestClient` that accepts a dummy `Application` that only implements methods, URLs, status codes & response.

And then in a test I'm using that `TestClient` instance to make requests (through an actual application). How can I verify if I'm passing correct payload to `POST` request? I've went through `TestClient`, `Application` &`TestServer`, but didn't find anything useful.

Should I just `mocker.spy` the `ClientSession`?

**Comment by Dreamsorcerer** [comment]
Have you got some sample code? I'm not overly clear on what you mean.
